File: SIGEVA-Service/src/main/java/com/equipo3/SIGEVA/model/Usuario.java
package com.equipo3.SIGEVA.model;

import java.util.Date;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import com.ulisesbocchio.jasyptspringboot.annotation.EncryptablePropertySource;

@Document
public class Usuario {
	@Id
	@Field
	private String idUsuario;
	@Field
	private String rol;
	@Field
	private String centroSalud;
	@Field
	private String username;
	@Field
	private String correo;
	@Field
	private String hashPassword;
	@Field
	private String dni;
	@Field
	@Value("${encryptedv.property}")
	private String nombre;
	@Field
	private String apellidos;
	@Field
	private Date fechaNacimiento;
	@Field
	private String imagen;

	public Usuario() {
		this.idUsuario = UUID.randomUUID().toString();
	}

	public String getIdUsuario() {
		return idUsuario;
	}

	public void setIdUsuario(String idUsuario) {
		this.idUsuario = idUsuario;
	}

	public String getRol() {
		return rol;
	}

	public void setRol(String rol) {
		this.rol = rol;
	}

	public String getCentroSalud() {
		return centroSalud;
	}

	public void setCentroSalud(String centroSalud) {
		this.centroSalud = centroSalud;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getCorreo() {
		return correo;
	}

	public void setCorreo(String correo) {
		this.correo = correo;
	}

	public String getHashPassword() {
		return hashPassword;
	}

	public void setHashPassword(String hashPassword) {
		this.hashPassword = hashPassword;
	}

	public String getDni() {
		return dni;
	}

	public void setDni(String dni) {
		this.dni = dni;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getApellidos() {
		return apellidos;
	}

	public void setApellidos(String apellidos) {
		this.apellidos = apellidos;
	}

	public Date getFechaNacimiento() {
		return fechaNacimiento;
	}

	public void setFechaNacimiento(Date fechaNacimiento) {
		this.fechaNacimiento = fechaNacimiento;
	}

	public String getImagen() {
		return imagen;
	}

	public void setImagen(String imagen) {
		this.imagen = imagen;
	}
}

File: SIGEVA-UI/src/app/Model/Usuario.ts
import {CentroSalud} from "./centro-salud";

export interface Usuario{
  rol: string;
  centroSalud: CentroSalud;
  username: string;
  correo: string;
  hashPassword:string;
  dni:string;
  nombre:string;
  apellidos:string;
  fechaNacimiento:string;
  imagen:string;
}

export class Usuario {
  rol: string;
  centroSalud: CentroSalud;
  username: string;
  correo: string;
  hashPassword:string;
  dni:string;
  nombre:string;
  apellidos:string;
  fechaNacimiento:string;
  imagen:string;

  constructor(rol: string, centroSalud: CentroSalud, username: string, correo: string, hashPassword:string, dni:string,
              nombre:string, apellidos:string, fechaNacimiento:string, imagen:string){

    this.rol = rol;
    this.centroSalud = centroSalud;
    this.username = username;
    this.correo = correo;
    this.hashPassword = hashPassword;
    this.dni = dni;
    this.nombre = nombre;
    this.apellidos = apellidos;
    this.fechaNacimiento = fechaNacimiento;
    this.imagen = imagen;
  }
}